### PR TITLE
e2e cleanup 20230628

### PR DESCRIPTION
**Pull request by ffromani**
further improvements and cleanups following the implementation of the NRT settlement support

### CODE REVIEW

**Review by swatisehgal — APPROVED** [review]
/lgtm

Changes look reasonable to me. I will defer whether we want to backport this or not to @shajmakh.

/hold
to avoid inadvertent merge.

**Comment by shajmakh** [comment]
Thank you for the cleanup.
/lgtm too.
up to you @ffromani to backport following your comment [here](https://github.com/openshift-kni/numaresources-operator/pull/626#issuecomment-1610816523)

**Comment by shajmakh** [comment]
Thank you for the cleanup.
/lgtm
up to you @ffromani to backport following your comment https://github.com/openshift-kni/numaresources-operator/pull/626#issuecomment-1610816523

File: test/utils/fixture/fixture.go
Comment (shajmakh) [inline test/utils/fixture/fixture.go:149]: I don't think we need this TODO because we're already waiting enough time for the NRT to equal to initial.

File: test/e2e/serial/config/infra.go
Comment (shajmakh) [inline test/e2e/serial/config/infra.go:113]: why? we're not using idx

File: test/utils/fixture/fixture.go
Comment (ffromani) [reply test/utils/fixture/fixture.go:154]: Not sure still. Probably yes.

File: test/utils/fixture/fixture.go
Comment (ffromani) [reply test/utils/fixture/fixture.go:149]: sounds good. Removing.

**Comment by ffromani** [comment]
the PR is now complete and ready for review

File: test/utils/fixture/fixture.go
Comment (ffromani) [reply test/utils/fixture/fixture.go:151]: The variable is initialized outside the `k8swait.PollImmediate` and captured by its closure, so it should always contain the most recent Get value, isn't it?

**Comment by ffromani** [comment]
Once approved and merged, I plan to backport this PR down to 4.12, but not immediately. Let's see few runs of it against 4.14 before doing all the backport dance.

File: test/utils/fixture/fixture.go
Comment (shajmakh) [reply test/utils/fixture/fixture.go:151]: I see the variable is declared but not initialized so it's zero-value will be nil. you are right, it will always contain the most recent value of Get/List but in case List kept failing the value will stay nil. Unless List do initialize an empty nrt-list in case of failure. I'm not sure about that, I'm interested to verify this.
Anyhow,its possibility to fail is close to zero, so we can keep it as is.

Comment (ffromani) [reply test/utils/fixture/fixture.go:151]: true, but if `currentNrtList` is zero-value, then `err` will be != nil, so we clearly catch this case

File: internal/noderesourcetopology/equality_test.go
Comment (shajmakh) [inline internal/noderesourcetopology/equality_test.go:110]: I think it's nearly enough to test this case. 
just to clarify, don't we want usually to add negative case chack, like the expected to be false, just to verify the equality check also works for the inner items as expected?

Comment (ffromani) [reply internal/noderesourcetopology/equality_test.go:110]: good point. Yes we should add at least a negative case

**Comment by shajmakh** [comment]
Thanks for the beneficial enhancements!
/lgtm

**Comment by shajmakh** [comment]
Thanks for adding the test
/lgtm

**Pull request merged**
